src/Utilities/Compiler/Extensions/CharExtensions.cs

```diff
@@ -0,0 +1,9 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the MIT license.  See License.txt in the project root for license information.
+
+namespace Analyzer.Utilities.Extensions
+{
+    internal static class CharExtensions
+    {
+        public static bool IsASCII(this char c) => (uint)c <= '\x007f';
```

Comment:
Consider adding a link to the dotnet/runtime code: https://github.com/dotnet/runtime/blob/d1fc57ea18ee90aee8690697caed2b9f162409eb/src/libraries/System.Private.CoreLib/src/System/Char.cs#L91C28-L91C35